Airlock Construction Update

uraniummeltdown opened the pull request:
Airlock Painter no longer works on airlock assemblies (still works on airlocks though), removed its vars from door_assembly
Cleaned up some copypasta in door_assembly code
Airlock assemblies update their name more regularly during construction
Many new airlock types can be built (by hand and RCD) and painted
Vault door assemblies can be built with 8 plasteel
Glass and Mineral sheets are now able to be welded out of door assemblies rather than having to deconstruct the whole thing
Plasma glass sheets no longer work on door assemblies. It can be readded once someone makes a plasma glass overlay (like a purple version of the glass overlay)
Airlocks take their dir from the assembly if constructed from it
Thanks to @WJohn for the solid station2 (public) airlock and survival pod airlock overlay sprites

![airlockassemblies](https://user-images.githubusercontent.com/18234833/33310451-1ad0e1f8-d43b-11e7-854f-650531adca1b.png)
Pic is a bit outdated, highsec assemblies need plasteel now


I will make glass airlocks a subtype of their regular version in a separate PR after this



:cl:
add: Added MANY new types of airlock assembly that can be built with metal. Use metal in hand to see the new airlock assembly recipes.
add: Added new airlock types to the RCD and airlock painter
add: Vault door assemblies can be built with 8 plasteel, high security assemblies with 6 plasteel
add: Glass mineral airlocks are finally constructible. Use glass and mineral sheets on an airlock assembly in any order to make them.
tweak: Glass and mineral sheets are now able to be welded out of door assemblies rather than having to deconstruct the whole thing
del: Airlock painter no longer works on airlock assemblies (still works on airlocks)
fix: Titanium airlocks no longer have any missing overlays
/:cl:

Fixes https://github.com/tgstation/tgstation/issues/32392

CryoSpacewalk committed:
removed typetext and icontext vars from door assemblies, airlock painter no longer works on door assemblies
removed 'anchored = TRUE, state = 1' copypasta from door assemblies
door assemblies update their name more regularly during construction
added new airlock types to airlock painter on airlocks, and to the RCD
airlock assemblies built with metal are now in a list, you can build many more airlocks
vault door assemblies can be built with 4 plasteel
added a solid version of station2 airlocks, renamed to "public airlock"
glass and mineral sheets are now able to be welded out of door assemblies

CryoSpacewalk committed:
find n replace airlock/glass to airlock/glass_public in maps
door_assembly_glass to door_assembly_public

XDTM commented:
High security airlocks have a higher damage threshold, may want plasteel for those

CryoSpacewalk committed:
made highsec airlocks built with 6 plasteel, removed from RCD and airlock painter
vault doors built with 8 plasteel
removed redundant vars in airlock_types.dm
reverts the dumb rglass 'replacement' addition in door assembly construction

xxalpha commented:
I am not going to look at the code and just ask: did you fix the spaghetti & snowflake code?

uraniummeltdown commented:
@xxalpha 
I got rid of some of the constant duplication in door assembly code, there is still some duplication in that most glass airlocks aren't a subtype of the non-glass airlock. I got rid of the airlock painter function on door assemblies altogether because it was atrocious

WJohn commented:
This is... really unappealing to look at. But so is our general metal sheet construction screen. This needs a bit more organization, maybe some classes (medsci containing R&D, viro, medical. command containing vault, high sec, and command doors etc etc). Up to you to how to organize these, of course.

uraniummeltdown commented:
@WJohn It's not that bad man, keep in mind that this is a separate menu in the metal recipe list, it does not spam the metal recipe list when you use metal in hand. Before this PR we weren't even able to build any airlock assemblies except the standard one.

Bawhoppen commented:
So why even are you removing the painter working on the assemblies though?

uraniummeltdown commented:
@Bawhoppen Because now you can just make every one by hand. Also the painter made no sense why it could paint an airlock fo look radically different (high sec airlocks are not regular), was missing a ton of airlocks, was super old awful buggy code that it had to go

CryoSpacewalk committed:
moved airlock assembly defines to construction defines file
fixed incorrectly named shuttle airlock glass_closed and maint panel protected closed overlays
improved examination message for glass/minerals present in assembly

Bawhoppen commented:
@uraniummeltdown But it can still do that with normal airlocks you said?

uraniummeltdown commented:
@Bawhoppen The way it works with normal airlocks isn't as bad (still bad though) and since this PR changes a lot I'll think about removing them in another one

CryoSpacewalk committed:
airlocks take their dir from the assembly when built from assembly
removed "vertical" variant of survival pod airlock, unneeded
survival pod airlock assembly can have glass applied
solid and glass overlays for survival pod airlocks

Bawhoppen commented:
You should expand their functionality, not remove it

CryoSpacewalk committed:
OOPs some airlocks, removes a duplicate door assembly, removes some var edits on maps for a new glass type

CryoSpacewalk committed:
turn copied door assembly code into a proc
some feedback during door assembly construction
door assembly created airlock takes name from base_name if no custom name
moved plasma airlock proc to airlock_types.dm

AnturK commented on code/modules/shuttle/shuttle_rotate.dm:
```diff
@@ -105,10 +101,6 @@ If ever any of these procs are useful for non-shuttles, rename it to proc/rotate
 	params = NONE
 	return ..()
 
-/obj/machinery/door/airlock/survival_pod/shuttleRotate(rotation, params)
```
What was these about ?

uraniummeltdown replied:
Apparently the survival pod airlock would not rotate when it was inside a shuttle that rotated, the dirs on survival pod airlocks were all messed up, there used to be 2 survival pod airlocks a horizontal and vertical variant but all the dirs were put into a single icon a while ago and made into a single airlock
After removing the shuttleRotate stuff I tested it with the nuke ops ship and nav computer, both the airlock and assembly have unique icons for all 4 dirs and they rotate as you would expect them to

AnturK commented on code/game/objects/structures/door_assembly.dm:
```diff
@@ -703,38 +493,66 @@
-				new mineral_path(T, 2)
+				new /obj/item/shard(T)
+		if(mineral)
+			var/obj/item/stack/sheet/mineral/mineral_path = text2path("/obj/item/stack/sheet/mineral/[mineral]")
```
Any way to get rid of these while you're refactoring the whole thing ?

uraniummeltdown replied:
I will take a look at it and ask on coderbus over the next few days, no guarantees though as I'm kind of a noob coder

AnturK replied:
Alright, can wait for another PR though.

AnturK commented:
If you could list path changes for downstreams/maps in progress i'd be grateful.

uraniummeltdown commented:
@AnturK 
Survival pod airlocks have solid and glass versions: existing /survival_pod changed to /survival_pod/glass, /survival_pod/vertical to /survival_pod/glass{dir = 4}

AnturK commented:
Thank you!
